Kuroba/app/src/main/java/com/github/adamantcheese/chan/features/gesture_editor/Android10GesturesExclusionZonesHolder.kt

```diff
@@ -0,0 +1,236 @@
+                    ExclusionZonesJson(minScreenSize, maxScreenSize, newExclusionZones)
+            )
+            ChanSettings.androidTenGestureZones.set(json)
+
+            Logger.d(TAG, "Added zone ${zoneRect} for orientation ${orientation}")
+        }
+    }
+
+    fun removeZone(orientation: Int, attachSide: AttachSide) {
+        if (exclusionZones.isEmpty()) {
+            return
+        }
+
+        if (!exclusionZones.containsKey(orientation)) {
+            return
+        }
+
+        val exclusionZone = getZoneOrNull(orientation, attachSide)
+                ?: return
+
+        if (exclusionZones[orientation]!!.remove(exclusionZone)) {
+            val newExclusionZones = mutableListOf<ExclusionZoneJson>()
+
+            exclusionZones.forEach { (orientation, zones) ->
+                zones.forEach { zone ->
+                    zone.checkValid()
+
+                    newExclusionZones += ExclusionZoneJson(
+                            screenOrientation = orientation,
+                            attachSide = zone.attachSide.id,
+                            left = zone.left.toFloat(),
+                            right = zone.right.toFloat(),
+                            top = zone.top.toFloat(),
+                            bottom = zone.bottom.toFloat()
+                    )
+                }
+            }
+
+            val json = gson.toJson(
+                    ExclusionZonesJson(minScreenSize, maxScreenSize, newExclusionZones)
+            )
+            ChanSettings.androidTenGestureZones.set(json)
+
+            Logger.d(TAG, "Removed zone ${exclusionZone} for orientation ${orientation}")
+        }
+    }
+
+    fun fillZones(zones: MutableMap<Int, MutableSet<ExclusionZone>>, skipZone: ExclusionZone?) {
+        zones.clear()
+        deepCopyZones(zones)
+
+        if (skipZone != null) {
+            skipZone.checkValid()
+            zones[skipZone.screenOrientation]?.remove(skipZone)
+        }
+    }
+
+    private fun deepCopyZones(zones: MutableMap<Int, MutableSet<ExclusionZone>>) {
+        exclusionZones.forEach { (orientation, set) ->
+            if (!zones.containsKey(orientation)) {
+                zones[orientation] = mutableSetOf()
+            }
+
+            set.forEach { zone ->
+                zones[orientation]!!.add(zone.copy())
+            }
+        }
+    }
+
+    fun getZones(): MutableMap<Int, MutableSet<ExclusionZone>> {
+        return exclusionZones
+    }
+
+    fun getZoneOrNull(orientation: Int, attachSide: AttachSide): ExclusionZone? {
+        val zones = exclusionZones[orientation]
+                ?.filter { zone -> zone.attachSide == attachSide }
+                ?: emptyList()
+
+        if (zones.isEmpty()) {
+            return null
+        }
+
+        if (zones.size > 1) {
+            val zonesString = zones.joinToString(prefix = "[", postfix = "]", separator = ",")
+
+            throw IllegalStateException("More than one zone exists with the same orientation " +
```

Comment:
You're throwing an exception here, but it looks like if you just let the function continue on then there would be no problems? Maybe change this to just be an error log statement?

Reply:
Well this just means that something went wrong and I forgot to filter something somewhere because there should be only one zone per attach side. This exception is just a notification that something broke somewhere and needs to be investigated.